### PR TITLE
Getting user data and setting to the components/admin/MainHeader.vue

### PR DESCRIPTION
**Pull request by allanhal**
Resolves: #59 #81

### CODE REVIEW

File: store/index.js [hunk 1]
```diff
@@ -10,25 +10,54 @@ const parseCookies = (cookie, key) => {
...
+  logo: Cookie.get('logo'),
 })
 
 export const mutations = {
-  setAuth(state, auth) {
-    Cookie.set('auth', auth)
-    state.auth = auth
+  setUserData(state, userData) {
+    if (!userData) {
+      Cookie.remove('token')
```
Comment (edenilsonfe) [inline store/index.js]: I think you can create a method to delegate this responsibility: e.g. : `removeCookie`

File: store/index.js [hunk 2]
```diff
@@ -10,25 +10,54 @@ const parseCookies = (cookie, key) => {
...
+      Cookie.remove('id')
+      Cookie.remove('name')
+      Cookie.remove('email')
+      Cookie.remove('logo')
+    } else {
+      const { token, data: { id, name, email, logo } = {} } = userData || {}
+      Cookie.set('token', token)
```
Comment (edenilsonfe) [inline store/index.js]: Same here, e.g. : `setCookie`

File: store/index.js [hunk 3]
```diff
@@ -10,25 +10,54 @@ const parseCookies = (cookie, key) => {
...
+  email: Cookie.get('email'),
+  logo: Cookie.get('logo'),
 })
 
 export const mutations = {
-  setAuth(state, auth) {
-    Cookie.set('auth', auth)
-    state.auth = auth
+  setUserData(state, userData) {
+    if (!userData) {
```
Comment (edenilsonfe) [inline store/index.js]: applying the suggestions below, you can use this verification:
if (!userData)
    return removeCookie();
return setCookie();

File: store/index.js (hunk 1 above)
Comment (allanhal) [reply store/index.js]: done

File: store/index.js (hunk 3 above)
Comment (allanhal) [reply store/index.js]: done

File: store/index.js (hunk 2 above)
Comment (allanhal) [reply store/index.js]: done

File: store/index.js [hunk 4]
```diff
@@ -10,25 +10,54 @@ const parseCookies = (cookie, key) => {
...
+      Cookie.remove('token')
+      Cookie.remove('id')
+      Cookie.remove('name')
+      Cookie.remove('email')
+      Cookie.remove('logo')
+    } else {
```
[the remark this answers is not in the document]
Comment (eerison) [reply store/index.js]: That's better but I think you should apply @edenilsonfe suggestion!

File: store/index.js [hunk 5]
```diff
@@ -10,25 +10,62 @@ const parseCookies = (cookie, key) => {
...
+  email: Cookie.get('email'),
+  logo: Cookie.get('logo'),
 })
 
 export const mutations = {
-  setAuth(state, auth) {
-    Cookie.set('auth', auth)
-    state.auth = auth
+  setUserData(state, userData) {
+    if (!userData) {
```
Comment (edenilsonfe) [inline store/index.js]:
```suggestion
    if (!userData) {
        removeCookie();
        return;
    }

    setCookie();
```
```suggestion
    if (!userData) {
      removeCookie()
    } else {
       setCookie(state, userData)
    }
```

File: components/LoginForm.vue [hunk 6]
```diff
@@ -67,11 +65,11 @@ export default {
         this.loading = true
         const response = await this.$accountRepository.auth(credentials)
 
-        this.$store.commit('setAuth', response.token) // mutating to store for client rendering
```
[the remark this answers is not in the document]
Comment (allanhal) [reply components/LoginForm.vue]: Got the whole response an set to cookies, so we can use it later. Like the name that I used on the components/admin/MainHeader.vue

[the remark this answers is not in the document]
Comment (eerison) [reply components/LoginForm.vue]: I know, I was talking about the variable name `setAuth`

File: store/index.js [hunk 7]
```diff
@@ -10,25 +10,62 @@ const parseCookies = (cookie, key) => {
...
+  Cookie.remove('name')
+  Cookie.remove('email')
+  Cookie.remove('logo')
+}
+
+const setCookie = (state, userData) => {
+  const { token, data: { id, name, email, logo } = {} } = userData || {}
+  Cookie.set('token', token)
+  Cookie.set('id', id)
+  Cookie.set('name', name)
```
Comment (carlosxjm) [inline store/index.js]: Store the user information as an object using a unique key on cookie.

Comment (allanhal) [reply store/index.js]: image
Tried to get the auth response to JSON.stringify and got those %20 :/
You know a solution to this? If not, let's get that merged and changed later on.

File: components/LoginForm.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (allanhal) [reply components/LoginForm.vue]: done

**Pull request merged**